frontend/app/src/main/java/dk/scheduling/schedulingfrontend/components/DateRangePicker.kt

```diff
@@ -101,32 +101,38 @@ fun StandardDateRangePicker(
 
 class DateRange {
     constructor(startTime: Long, endTime: Long) {
-        rangeStart = startTime
-        rangeEnd = endTime
+        val startInstant = Instant.ofEpochMilli(startTime)
+        rangeStart = LocalDateTime.ofInstant(startInstant, ZoneId.systemDefault())
+        val endInstant = Instant.ofEpochMilli(endTime)
+        rangeEnd = LocalDateTime.ofInstant(endInstant, ZoneId.systemDefault())
     }
     constructor() {
         rangeStart = null
         rangeEnd = null
     }
 
-    private val rangeStart: Long?
-    private val rangeEnd: Long?
+    private val rangeStart: LocalDateTime?
+    private val rangeEnd: LocalDateTime?
+    private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
 
     fun isValidRange(): Boolean {
-        return rangeStart != null && rangeEnd != null && rangeStart <= rangeEnd
-    }
-
-    private fun convertMillisToDate(millis: Long): String {
-        val formatter = SimpleDateFormat("dd/MM/yyyy", Locale.GERMANY)
-        return formatter.format(Date(millis))
+        return rangeStart != null && rangeEnd != null && (rangeStart.isBefore(rangeEnd) || rangeStart.isEqual(rangeEnd))
     }
 
     fun getStartDate(): String {
-        return if (isValidRange()) convertMillisToDate(rangeStart!!) else "No start date"
+        return if (rangeStart != null) rangeStart.format(formatter) else "No start date"
     }
 
     fun getEndDate(): String {
-        return if (isValidRange()) convertMillisToDate(rangeEnd!!) else "No end date"
+        return if (rangeEnd != null && rangeStart != null) {
+            if (rangeStart.isBefore(rangeEnd) || rangeStart.isEqual(rangeEnd)) {
+                rangeEnd.format(formatter)
+            } else {
+                "Start date must be before end date"
+            }
+        } else {
+            "No end date"
+        }
```

Comment:
Maybe shorten the function as since the distinction of "No end date" and "Invalid" is unnecessary "Invalid" -> "No or start is after end".
```suggestion
        return if (isValidRange()) {
                rangeEnd.format(formatter)
            } else {
                "No valid end date"
            }
```

Reply:
The users would be confused as they would not know why the end date is invalid.

Reply:
Why can you not do the same way as the getStart?